### PR TITLE
Implement YAML Based Shuffle Test

### LINKED ISSUE #738: Run Shuffle Tests Using the YAML Test Parser and Simulated Backend
With the merge of #681, we can utilize the yaml test parser for shuffle tests mentioned [here](https://github.com/ethereum/eth2.0-specs/blob/master/specs/test-format.md). Having deterministic shuffling across all client implementations is crucial. This issue can be closed once we successfully run a [shuffle test](https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/utils/shuffle.go) using the yaml test parser.

### PR DESCRIPTION
**Pull request by houjieth**
This resolves #738 

---

# Description

Added validator shuffle tests based Raul's prior work on yaml-based test case specifications.

### CODE REVIEW

File: beacon-chain/chaintest/yaml_test.go [hunk 1]
```diff
@@ -1,44 +1,39 @@
...
-	for _, tt := range chainTests {
-		for _, testCase := range tt.TestCases {
-			if err := sb.RunChainTest(testCase); err != nil {
-				t.Errorf("Beacon Chain test failed: %v", err)
-			}
-		}
+	chainTest := &backend.ChainTest{TestCases: []*backend.ChainTestCase{chainTestCase}}
+	shuffleTest := &backend.ShuffleTest{TestCases: []*backend.ShuffleTestCase{shuffleTestCase}}
+	if err = runTests([]interface{}{chainTest, shuffleTest}, sb); err != nil {
+		log.Fatalf("Fail to run test cases: %v", err)
```
Comment (nisdas) [inline beacon-chain/chaintest/yaml_test.go]: It should be `Failed` instead of `Fail`

File: beacon-chain/chaintest/yaml_test.go [hunk 2]
```diff
@@ -1,44 +1,39 @@
...
 )
 
-func TestYamls(t *testing.T) {
-	sb, err := backend.NewSimulatedBackend()
-	if err != nil {
-		t.Fatalf("Could not create backend: %v", err)
+// TestReadTestsFromYaml tests constructing test cases from yaml file
+func TestReadTestsFromYaml(t *testing.T) {
+	if _, err := readTestsFromYaml(string("./tests")); err != nil {
+		t.Fatalf("Failed to read yaml files")
```
Comment (rauljordan) [inline beacon-chain/chaintest/yaml_test.go]: make sure to add the err to the fatalf call

File: beacon-chain/chaintest/yaml_test.go [hunk 3]
```diff
@@ -1,44 +1,39 @@
...
 
-	var chainTests []*backend.ChainTest
-
-	files, err := ioutil.ReadDir("./sampletests")
+// TestReadTestsFromYaml tests the running of provided tests structs
+func TestRunTests(t *testing.T) {
+	sb, err := backend.NewSimulatedBackend()
 	if err != nil {
-		t.Fatalf("Could not read yaml tests directory: %v", err)
+		log.Fatalf("Could not create backend: %v", err)
```
Comment (rauljordan) [inline beacon-chain/chaintest/yaml_test.go]: why was this changed to log.fatalf ? instead of t.Fatalf?
